### PR TITLE
Update MacOS setup instructions

### PR DESCRIPTION
**Pull request by gpeng**
Fix some spelling errors/typos and add asset setup instructions to the setup-osx docs.

Also update ruby version instructions (2.2.8 -> 2.5.0)

### CODE REVIEW

File: docs/setup-osx.md [hunk 1]
```diff
@@ -166,6 +166,16 @@ bundle install
...
 ```
 
+#### Configure assets
+
+```shell
+npm install -g yarn
+```
+
+```shell
+bundle exec rake tmp:create yarn:install assets:precompile
```
Comment (yivo) [inline docs/setup-osx.md]: `assets:precompile` and `tmp:create` are not required here

**Review by yivo — CHANGES_REQUESTED** [review]

**Comment by yivo** [comment]
@gpeng Thank you for the contribution. Could you pls push minor improvements and squash commits?

File: docs/setup-osx.md (hunk 1 above)
Comment (gpeng) [reply docs/setup-osx.md]: Thanks @yivo fixed in b643194

**Review by mod — APPROVED** [review]
Many Thanks @gpeng

**Pull request merged**